tslearn/barycenters/softdtw.py

```diff
@@ -100,14 +100,22 @@ def softdtw_barycenter(X, gamma=1.0, weights=None, method="L-BFGS-B", tol=1e-3,
         barycenter = init
 
     if max_iter > 0:
-        X_ = numpy.array([to_time_series(d, remove_nans=True) for d in X_])
+        X_ = numpy.array(
+            [to_time_series(d, remove_nans=True) for d in X_], dtype=object
```

Comment:
Why `dtype=object` here ?

Reply:
One of the error messages was:

```
=================================== FAILURES ===================================
___________ [doctest] tslearn.barycenters.softdtw.softdtw_barycenter ___________
069     Examples
070     --------
071     >>> time_series = [[1, 2, 3, 4], [1, 2, 4, 5]]
072     >>> softdtw_barycenter(time_series, max_iter=5)
073     array([[1.25161574],
074            [2.03821705],
075            [3.5101956 ],
076            [4.36140605]])
077     >>> time_series = [[1, 2, 3, 4], [1, 2, 3, 4, 5]]
078     >>> softdtw_barycenter(time_series, max_iter=5)
UNEXPECTED EXCEPTION: ValueError('setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.')
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.17/x64/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest tslearn.barycenters.softdtw.softdtw_barycenter[3]>", line 1, in <module>
  File "/home/vsts/work/1/s/tslearn/barycenters/softdtw.py", line 103, in softdtw_barycenter
    X_ = numpy.array([to_time_series(d, remove_nans=True) for d in X_])
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
/home/vsts/work/1/s/tslearn/barycenters/softdtw.py:78: UnexpectedException

```

Running the code on my local computer, I had the following error message:

```
/usr/bin/python3.8 /home/ycabanes/work/tslearn/codes/tslearn_applications/test_barycenters_soft_dtw.py 
/home/ycabanes/work/tslearn/codes/tslearn/tslearn/barycenters/softdtw.py:103: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  X_ = numpy.array([to_time_series(d, remove_nans=True) for d in X_])
```

Reply:
From what I understand, creating an numpy array with inhomogeneous shape is deprecated.
We should specify `dtype=object` if we want to make it possible.

Reply:
What should be done in your opinion @rtavenar ?

Reply:
That's not what we want I think, not sure why it's done this way here. What if we use https://tslearn.readthedocs.io/en/latest/gen_modules/utils/tslearn.utils.to_time_series_dataset.html ?